packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/SingleImageViewer/SingleImageViewerContainer.js

```diff
@@ -59,14 +48,22 @@ class SingleImageViewerContainer extends React.Component {
         loading: asyncStates.success
       })
     } catch (error) {
-      console.error(error)
-      this.setState({ loading: asyncStates.error })
+      this.onError(error)
     }
   }
 
+  onLoad (event) {
+    this.props.onReady(event)
```

Comment:
Does the event object need to be passed along here? The `onSubjectReady` store function is just setting a boolean and doesn't do anything with the event.

Reply:
I originally thought `onReady` would be a proxy handler for the DOM's `load` event but that's only true if the store ever needs to reflect on the event eg. to get info about the downloaded image file. The image loading code is slightly complicated anyway, because of the preloader that I'm assuming is there to get the images original height and width?

Reply:
Yes, the original PFE code gets the original height and width as well as the client height and width  at the onload event and stores them in the classification metadata. We still need to add that data to the classification metadata as noted in #304, but it's hasn't been done yet because we're now using an svg `image` instead of an html `img` and it doesn't have a client height and width. I'm not sure if we should go forward with the change to svg `image` or not.

Reply:
I could see this being used to store the image's original height and width in the subject viewer store and the classification store could observe for that state and update the metadata.

Reply:
Yeah, if you inspect the `image` element in the DOM, client height and width are 0 in Chrome. Although we've recorded those numbers historically, I don't think they've ever been used in data analysis. Caesar doesn't need the client size, as far as I know, although it will need the natural size of an SVG image in order to analyse drawing annotations.

>I'm not sure if we should go forward with the change to svg image or not.

I don't think you can have drawing tasks or zoom and rotation transforms without using SVG.

Reply:
PFE renders an SVG rectangle just to get the client bounding box for drawing tasks:
https://github.com/zooniverse/Panoptes-Front-End/blob/fe0c87a8ef589ef6c662bbb350ed35f8b1b8e895/app/classifier/annotation-renderer/svg.jsx#L18-L31

Anti-Slavery Manuscripts is a drawing task on top of a jpeg image, so it might be worth looking at their code to see how they get the image dimensions in oder to map drawn points to positions on a page. They might have used a better approach than the one in PFE. AnnoTate might be worth looking at too.

Reply:
I've started work on this in #804.

Reply:
If we do need client height/width, they can be read from one of the containing DOM elements. Or we can get the client bounding box of the SVG element. #804 is pretty much done but I'm recording the client dimensions as 0 at the moment.